### PR TITLE
[WiFILED] Use DS annotations, fix static code analysis findings

### PR DESCRIPTION
**Pull request by wborn**
* Use DS annotations
* Fix static code analysis findings
* Add org.openhab.binding.wifiled.test to build

Are you OK with these changes @xylo, @rvt?

### CODE REVIEW

File: addons/binding/org.openhab.binding.wifiled.test/pom.xml [hunk 1]
```diff
@@ -1,64 +1,64 @@
...
+		<artifactId>pom</artifactId>
+		<version>2.3.0-SNAPSHOT</version>
+	</parent>
 
-    <properties>
-        <bundle.symbolicName>org.openhab.binding.wifiled.test</bundle.symbolicName>
-        <bundle.namespace>org.openhab.binding.wifiled.test</bundle.namespace>
-    </properties>
+	<properties>
+		<bundle.symbolicName>org.openhab.binding.wifiled.test</bundle.symbolicName>
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.wifiled.test/pom.xml]: Please just remove these 4 lines

File: addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/FadingWiFiLEDDriver.java [hunk 2]
```diff
@@ -42,22 +42,23 @@
     public static final int DEFAULT_FADE_STEPS = 100;
     public static final int KEEP_COMMUNICATION_OPEN_FOR_MS = 1000;
 
-    private static final InternalLedState blackState = new InternalLedState();
+    private static final InternalLedState BLACK_STATE = new InternalLedState();
 
     private boolean power = false;
-    private InternalLedState currentState = new InternalLedState(); // USe to not update the controller with the same value
+    private InternalLedState currentState = new InternalLedState(); // USe to not update the controller with the same
+                                                                    // value
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/FadingWiFiLEDDriver.java]: Maybe correct the typo Use

File: addons/binding/pom.xml [hunk 3]
```diff
@@ -110,6 +110,7 @@
 		<module>org.openhab.binding.urtsi</module>
 		<module>org.openhab.binding.vitotronic</module>
 		<module>org.openhab.binding.wifiled</module>
+		<module>org.openhab.binding.wifiled.test</module>
```
Comment (martinvw) [inline addons/binding/pom.xml]: Did you check whether it was ever included but removed? Anyone else some memory about this?

Comment (wborn) [reply addons/binding/pom.xml]: It seemed to be missing from https://github.com/openhab/openhab2-addons/pull/2496 by @rvt which was merged yesterday.

**Comment by martinvw** [comment]
Thanks!

**Comment by xylo** [comment]
I'm OK with these changes.

**Comment by martinvw** [comment]
Sorry I intended to wait but forgot, thanks for checking

**Comment by xylo** [comment]
No problem.  Code style improvements are always fine for me and I guess that @rvt is also OK with it.

**Comment by rvt** [comment]
I am good.

**Pull request merged**